extensions/wikia/PowerUser/maintenance/PowerUserPopulatePropertiesMaintenance.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Maintenance script for populating user_properties tables
+ * with initial data on power users.
+ *
+ * @package MediaWiki
+ * @addtopackage maintenance
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ *
+ *
+ */
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+use Wikia\PowerUser\PowerUser;
+
+class PowerUserPopulatePropertiesMaintenance extends Maintenance {
+
+	private
+		$iPowerUsersLifetimeCounter = 0,
+		$iPowerUsersAdminCounter = 0;
+
+	/**
+	 * Workflow:
+	 * 1. Populate the table with PowerUser properties of a lifetime type
+	 * 2. Populate the table with PowerUser properties of an admin type
+	 */
+	public function execute() {
+		$this->output( "Populating with PowerUsers for lifetime edits... \n" );
+		$this->populatePowerUsersLifetime();
+		$this->output( "Populating with PowerUsers for admin rights... \n" );
+		$this->populatePowerUsersAdmin();
+	}
+
+	private function populatePowerUsersLifetime() {
+		global $wgExternalSharedDB;
+		$oDB = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$aPowerUsersLifetimeIds = ( new WikiaSQL() )
+			->SELECT( 'user_id' )
+			->FROM( 'user' )
+			->WHERE( 'user_editcount' )->GREATER_THAN_OR_EQUAL( PowerUser::MIN_LIFETIME_EDITS )
+			->runLoop( $oDB, function( &$aPowerUsersLifetimeIds, $oRow ) {
+				$aPowerUsersLifetimeIds[] = $oRow->user_id;
+			} );
+
+		foreach ( $aPowerUsersLifetimeIds as $iUserId ) {
```

Comment:
You're running a loop with the DB query above and just creating this array to loop over again. Any reason not to just put this logic inside the `runLoop` so you only go over it once? i.e.

```php 
->runLoop( $oDB, function( $data, $oRow ) {
	$oPowerUser = new PowerUser( User::newFromId( $oRow->user_id ) );
	if ( $oPowerUser->addPowerUserProperty( PowerUser::TYPE_LIFETIME ) ) {
		$this->iPowerUsersLifetimeCounter++;
	}
} );